### DISCUSSION TITLE
Unable to read Cookie parameters

### DISCUSSION THREAD

**Discussion by xplato** [fastapi/fastapi#6432] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [x] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Cookie

@app.get("/me")
async def get_me(authtoken: str = Cookie(None)):
	print("TOKEN", authtoken) # Returns: TOKEN: None
	return {
		"auth_token": authtoken # Returns None
	}
```

### Description

(Note: the TL;DR at the bottom summarizes the most common issues with Cookie handling in FastAPI, and how my issue does not match these common issues.)

I have been on this issue for some time. Essentially, _FastAPI isn't reading the cookies from the request object._

My setup is NextJS + FastAPI. I've been able to confirm that FastAPI is able to **set** cookies in the response, and the front-end reflects this. I have also been able to confirm that **subsequent requests from the front-end do indeed send these cookies to FastAPI.** (Via logging `context.req.cookies`, meaning NextJS is properly sending them. Moreover, using `axios`, I set `withCredentials: true`, and this does work properly.)

Following the suggestion from [this issue](https://github.com/tiangolo/fastapi/issues/4029), I've confirmed that my cookie's name and what I'm using to reference the cookie in FastAPI is the same (`authtoken`).

I've confirmed that setting `secure=False, samesite="lax"`, etc. does indeed work properly, and the browser console reflects this information properly.

I've added CORSMiddleware with:

```
allow_origins=origins,
allow_credentials=True,
allow_methods=["*"],
allow_headers=["*"]
```

Note: this is an issue with localhost. I have not tested it on the live server because I can't even get this working.

---

The Cookie Parameters documentation within FastAPI is so small, meaning it should be a very simple thing. Unfortunately, however, it's not being so for me.

I have no clue what's going wrong here. I've spent a long time searching for issues, Stack Overflow answers, etc. all to no avail.

Thank you all so much for your time.

### Operating System

macOS

### Operating System Details

MacOS Monterey. Does not work on Safari, Firefox, Firefox Developer Edition, or Brave.

### FastAPI Version

0.70.0

### Python Version

3.10.0

### Additional Context

My `axios` request data within [`getServerSideProps`](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering):

```
{
	withCredentials: true,
	crossDomain: true,
	headers: {
		'Cookies': 'auth_token',
		'Access-Control-Allow-Origin': '*',
		'Content-Type': 'application/json'
	},
}
```

Tl;DR: I have confirmed the following:

- Cookies **are** being **set** by FastAPI properly.
- All browsers listed above **are** showing the cookie present.
- The request made from NextJS to the `/me` endpoint **does** include the cookies in the request.
- Moreover, within the Network tab of each of the browsers above, I've confirmed that each individual request does indeed sent request cookies.
- The name I use to set, reference, and send the cookie within both FastAPI and NextJS is the **same**: `authtoken`

**Comment by xplato** [comment, marked as answer]
This was not a FastAPI error. It was because NextJS's `getServerSideProps` function doesn't automatically attach cookies to requests inside of it. These must be passed manually.
